src/integrationTest/java/uk/gov/hmcts/reform/pip/rules/controllers/ControllerTest.java

```diff
@@ -35,67 +31,71 @@ public void welcomeRootEndpoint() throws Exception {
 
         ObjectMapper objectMapper = new ObjectMapper();
         objectMapper.registerModule(new JavaTimeModule());
-        Publication publication = objectMapper.readValue(
-            response.getResponse().getContentAsByteArray(), Publication.class);
+        Court[] courts = objectMapper.readValue(response.getResponse().getContentAsByteArray(), Court[].class);
 
-        final int publicationID = 1;
-        assertEquals(publicationID, publication.getPublicationId(), "Publication ID is " + publicationID);
-        assertEquals(1, publication.getCourtHearingsList().size(), "The array contains a single publication");
+        final int courtID = 1;
+        assertEquals(courtID, courts[0].getCourtId(), "Court ID is " + courtID);
+        assertEquals(13, courts[0].getHearingList().size(), "The array contains a list of hearings");
     }
 
-    @DisplayName("Should return a 200, with a publication that contains a single court list")
+    @DisplayName("Should return a 404, with no courts")
     @Test
-    public void publicationThatExistsHearingsListTest() throws Exception {
-        MvcResult response = mockMvc.perform(get("/publication/1")).andReturn();
+    public void courtListDoesNotExistTest() throws Exception {
+        MvcResult mvcResult = mockMvc.perform(get("/courtlist/foobar")).andReturn();
+        assertEquals(404, mvcResult.getResponse().getStatus(), "Check that the status is a 404");
+    }
+
+    @DisplayName("Should return a 200, and all court list ")
+    @Test
+    public void courtListAllTest() throws Exception {
+        MvcResult response = mockMvc.perform(get("/courtlistall/")).andReturn();
 
         assertEquals(200, response.getResponse().getStatus(),
                      "Check that a 200 status is returned is returned");
 
         ObjectMapper objectMapper = new ObjectMapper();
         objectMapper.registerModule(new JavaTimeModule());
-        Publication publication = objectMapper.readValue(
-            response.getResponse().getContentAsByteArray(), Publication.class);
+        Court[] courts = objectMapper.readValue(response.getResponse().getContentAsByteArray(), Court[].class);
 
-        List<CourtHearings> courtHearingsList = publication.getCourtHearingsList();
-        assertThat(courtHearingsList.get(0).getCourtId()).as("Get first court id").isEqualTo(1);
+        final int courtID = 1;
+        assertEquals(courtID, courts[0].getCourtId(), "Court ID is " + courtID);
+        assertEquals(583, courts.length, "The array contains a list of courts");
     }
 
-    @DisplayName("Should return a 200, with a publication that contains a multiple hearings")
+    @DisplayName("Should return a 200, and all hearing list for the court id passed and the today date")
     @Test
-    public void publicationThatExistsHearingsTest() throws Exception {
-        MvcResult response = mockMvc.perform(get("/publication/1")).andReturn();
+    public void hearingListForTodayDateTest() throws Exception {
```

Comment:
This test appears to be failing. It's returning a 404 rather than a 200.

Reply:
test fixed